api/app/src/test/kotlin/packit/integration/controllers/PacketControllerTest.kt

```diff
@@ -13,78 +12,40 @@ import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import packit.integration.IntegrationTest
 import packit.integration.WithAuthenticatedUser
-import packit.model.Packet
-import packit.model.PacketGroup
-import packit.model.dto.PacketDto
 import packit.repository.PacketGroupRepository
 import packit.repository.PacketRepository
+import packit.service.PacketService
 import kotlin.test.assertEquals
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class PacketControllerTest : IntegrationTest()
 {
+    @Autowired
+    private lateinit var packetService: PacketService
+
     @Autowired
     private lateinit var packetRepository: PacketRepository
 
     @Autowired
     private lateinit var packetGroupRepository: PacketGroupRepository
-    private lateinit var packets: List<Packet>
-    private lateinit var packetGroups: List<PacketGroup>
 
     companion object {
-        const val idOfArtefactTypesPacket1 = "20240729-154633-10abe7d1"
-        const val idOfArtefactTypesPacket2 = "20240729-155513-1432bfa7"
+        const val idOfArtefactTypesPacket = "20240729-154633-10abe7d1"
         const val idOfComputedResourcePacket = "20240729-154635-88c5c1eb"
-        const val hashOfReport = "715f397632046e65e0cc878b852fa5945681d07ab0de67dcfea010bb6421cca1"
+        const val hashOfReport = "sha256:715f397632046e65e0cc878b852fa5945681d07ab0de67dcfea010bb6421cca1"
     }
 
     @BeforeAll
     fun setupData()
     {
-        packets = packetRepository.saveAll(
-            listOf(
-                Packet(
-                    idOfArtefactTypesPacket1,
-                    "artefact-types",
-                    "artefact-types",
-                    emptyMap(),
-                    false,
-                    0.0,
-                    0.0,
-                    0.0
-                ),
-                Packet(
-                    idOfArtefactTypesPacket2,
-                    "artefact-types",
-                    "artefact-types",
-                    emptyMap(),
-                    false,
-                    0.0,
-                    0.0,
-                    0.0
-                ),
-                Packet(
-                    idOfComputedResourcePacket,
-                    "computed-resource",
-                    "computed-resource",
-                    emptyMap(),
-                    false,
-                    0.0,
-                    0.0,
-                    0.0
-                ),
-            )
-        )
-        packetGroups = packetGroupRepository.saveAll(
-            packets.map { it.name}.distinct().map{PacketGroup(it)}
-        )
+        packetService.importPackets()
```

Comment:
Sorry, I can't quite remember how these integration tests were set up. Do the ids defined above happen to exist in the demo dataset so just doing importPackets here suffices rather than stuffing the packets into the repo as before?

Reply:
Yes. This test was doing a mix of using the files that are on disk and inserting a bunch of fake entries in the database.

Any test that leads to requests to outpack server will need the files to exist in the repository anyway, so it seems simpler to just use that consistently and not try to setup these fake packets that don't exist on disk. Ideally we'd create those packets programmatically in the test setup instead of checking them in to Git, but there is nothing in Packit that is capable of doing that.